File: 0046-permutations/0046-permutations.cpp
class Solution {
public:
    vector<vector<int>> permute(vector<int>& nums) {
        
      vector<vector<int>>ans;
      
      find(0,nums,ans);
      return ans;
    }
  
     void find(int start,vector<int>&nums,vector<vector<int>>&ans){
       if(start==nums.size()){
         
         ans.push_back(nums);
           
         return;
       }
       
          for(int i=start;i<nums.size();i++){
           swap(nums[start],nums[i]);
            find(start+1,nums,ans);
            swap(nums[start],nums[i]);
         }
     }
    
 };

File: 0046-permutations/0046-permutations.py
class Solution:
    def permute(self, nums: List[int]) -> List[List[int]]:
        
        
        ans=[]
        
        n=len(nums)
        
        self.find(0,n,nums,ans)
        
        return ans
    
    def find(self,i,n,nums,ans):
        
        if i >= n:
            ans.append(nums[:])
            return
        
        for j in range(i,n,1):
            nums[i],nums[j]=nums[j],nums[i]
            self.find(i+1,n,nums,ans)
            nums[i],nums[j]=nums[j],nums[i]
        
        